### PR TITLE
 Add testcase for Apim throttling custom extension in ballerina streams

### PR DESCRIPTION
**Pull request by gimantha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import ballerina/streams;
+import ballerina/time;
+
+public type TimeBatch object {
+    public string EXPIRY_TIME_STAMP;
```
Comment (mohanvive) [inline stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal]: why the variable name is defined as capital characters?

File: stdlib/streams/src/test/resources/test-src/streamingv2-custom-extension-test3.bal [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    StockRecord s3  = { symbol: "IBM", price: 700.0, volume: 0 };
+
+    deployThrottleQuery();
+
+    outputStream.subscribe(function(StockSumRecord e) {printStockDetails(e);});
+
+    inputStream.publish(s1);
+    runtime:sleep(500);
+    inputStream.publish(s2);
+    runtime:sleep(260000);
```
Comment (mohanvive) [inline stdlib/streams/src/test/resources/test-src/streamingv2-custom-extension-test3.bal]: can we reduce the sleep time (and also window time) for this test case? If not, the test case alone takes a considerable amount of time.

File: stdlib/streams/src/main/ballerina/streams/aggregators.bal [hunk 3]
```diff
@@ -34,28 +34,32 @@ public type Sum object {
...
+                if (eventType == "CURRENT") {
+                    self.fSum += value;
+                } else if (eventType == "EXPIRED"){
+                    self.fSum -= value;
+                } else if (eventType == "RESET"){
+                    self.fSum = 0.0;
+                }
+                return self.fSum;
+            } else if (value is ()) {
+                //do nothing
```
[the remark this answers is not in the document]
Comment (gimantha) [reply stdlib/streams/src/main/ballerina/streams/aggregators.bal]: If the event type is "reset", values will not be there. then the value is ()

File: stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import ballerina/streams;
```
[the remark this answers is not in the document]
Comment (gimantha) [reply stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal]: fixed

File: stdlib/streams/src/test/resources/test-src/streamingv2-custom-extension-test3.bal (hunk 2 above)
Comment (gimantha) [reply stdlib/streams/src/test/resources/test-src/streamingv2-custom-extension-test3.bal]: fixed

File: stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal (hunk 1 above)
Comment (gimantha) [reply stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal]: it has to be a const.. will fix

Comment (gimantha) [reply stdlib/streams/src/test/resources/test-src/throttler/time-batch.bal]: it seems we cannot define const in objects.. so i changed it to a field variable

**Pull request merged**